### ISSUE TITLE
Support for original FMC block diagrams

### ISSUE THREAD

**Issue by koppor** [oalt/fmc4se#2]
> An einer Stelle bin ich bei der Umsetzung der FMC-Profils von der in der Literatur beschriebenen Form abgewichen: Bei FMC werden Kommunikationskanäle durch einen Konnektor mit einem Kreis in der Mitte dargestellt. Ich nutze hingegen den Kreis bei Quelle und Ziel des Kommunikationskanals analog zu einem Port in UML oder SysML. [[source](http://mdd4all.de/fmc4se/)]

Would it be possible to enable FMC modeling as defined originally (see https://github.com/f-m-c/information)? I personally like the concept of the channel as is and also refer to the standard while promoting FMC. Finally, Prof. Wendt strongly argues for the separation. See <http://www.alternative-softwaresystemtechnik.de/FMC/FMC3.html> for details.

Oliver Kopp <oliver.kopp@mbition.io>, Mercedes-Benz AG on behalf of [MBition GmbH](https://mbition.io/).

[Imprint](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md#mbition-gmbh)

**Comment by oalt** [comment]
Hi!
Sorry for the late reply, but I saw the issue just today. 
I have added a new element representing a 'classic' FMC channel. The changes are available in the MDG technology 1.6 and the plugin. Try it out if it is helpfull for your application scenario.

Best regards

Oliver Alt

**Comment by oalt** [comment]
The plugin installer, supporting the explicit channel is available here: https://dev.azure.com/oa0170/_apis/resources/Containers/8409949/Output?itemPath=Output%2FFMC4SE-Plugin-Setup_3.4.0.223.msi

**Comment by koppor** [comment]
Thank you for the follow-up! I could load the plugin, but I am still struggling howto create a FMC diagram. 😇 I can create a diagram using a wizard, but not in other ways (using EA only very seldomly).

In other words, "und dann nutzen" would need to be extracted for me.

![grafik](https://user-images.githubusercontent.com/1366654/109552136-b9ddb380-7ad1-11eb-99e1-c9b39634ed3e.png)

**Comment by koppor** [comment]
Update - I found it. Will check which of these two it is:

![image](https://user-images.githubusercontent.com/1366654/109784758-acc4df80-7c0b-11eb-8444-619c4e97db04.png)

**Comment by koppor** [comment]
@oalt Could you please upload a new version to the [release page](https://github.com/oalt/fmc4se/releases)?

At the other resource, I get a 404 of azure (anonymized)

```json
{
  "$id": "1",
  "innerException": null,
  "message": "TF400813: The user '9652d7c2-xxxxx-4940-8151-4a92bd474ed0\\u@domain' is not authorized to access this resource.",
  "typeName": "Microsoft.TeamFoundation.Framework.Server.UnauthorizedRequestException, Microsoft.TeamFoundation.Framework.Server",
  "typeKey": "UnauthorizedRequestException",
  "errorCode": 0,
  "eventId": 3000
}
```

**Comment by koppor** [comment]
May I ping regarding this? Although I would really like to try to improvement, I still cannot download the updated version of the plugin.

![image](https://user-images.githubusercontent.com/1366654/118887876-3c208b80-b8fb-11eb-8c4d-41cfe87889aa.png)

![image](https://user-images.githubusercontent.com/1366654/118887896-45115d00-b8fb-11eb-8b26-d70f490752b4.png)

**Comment by oalt** [comment]
Hello @koppor !

What is the problem with the download? When you download the .msi and install it you will get the plugin and the technology installed on your PC. The technology is copied to `CommonAppDataFolder`/MDD4All/FMC4SE - perhaps I will change this in the future, because this is per default a hidden folder in the Explorer.

You can download the MDG technology as standalone file under: https://raw.githubusercontent.com/oalt/fmc4se/master/MDG_Technology/FMC4SE_MDG.xml

I test the GitHub download without being logged in. Try to download it from here - not from Azure directly: https://github.com/oalt/fmc4se/releases/tag/release%2F3.3.0.221 (Assets)

Best regards

@oalt

**Comment by koppor** [comment]
> What is the problem with the download? When you download the .msi 

In short: I am unable to download the MSI for 223.

Long:

I try to unpack my thoughts. Thereby, I try to avoid back references. I will put screenshots only as supporting material. I hope the text is clear enough to be understandable without images. I will try to make my assumptions clear and to assuring the assumptions by questions. Hope, in that way, I find a solution to use the **classic FMC channel**.

My aim is to get an `.msi` file as I could not manage to create a `.msi` for myself.

As far as I understood, the fix is contained in build 3.4.**0223** and not in build 3.4.**0221**. Thereby, I assume that 0223 is greater than 0221 and indicates that new features come in later versions.

In the release page, I can only download 0221.

![image](https://user-images.githubusercontent.com/1366654/118887896-45115d00-b8fb-11eb-8b26-d70f490752b4.png)

If I understood your comment correctly, you are pointing to 0223.

![image](https://user-images.githubusercontent.com/1366654/123074987-16196a00-d418-11eb-8cfb-43a68a6bafca.png)

Questions:

- Am I right that the release 0223 is newer than 0221?
- Am I right that the release 0221 does not contain the classic FMC channel?
- Am I right that the release 0223 contains the classic FMC channel feature?
- Am I right that the release page shows 0221 only?
- Am I right that the only way to access the binary for 0223 is the link dev.azure.com/oa0170/_apis/resources/Containers/8409949/Output?itemPath=Output%2FFMC4SE-Plugin-Setup_3.4.0.223.msi?

**Comment by koppor** [comment]
@oalt May I ping you regarding this? I still cannot find the 223 binary.

![grafik](https://user-images.githubusercontent.com/1366654/129938096-9a7fd75c-724a-444d-9e77-4cbe7d7b6c15.png)

**Comment by oalt** [comment]
@koppor : You can use the 221 version. I think it will support everything what you need https://github.com/oalt/fmc4se/releases/download/release%2F3.3.0.221/FMC4SE-Plugin-Setup_3.3.0.221.msi

Best regards

**Comment by koppor** [comment]
I Tried again today:

1. Installed the component  
  ![image](https://user-images.githubusercontent.com/1366654/136580931-3fbdbe18-cd7d-41a9-aa0f-ebd3af93b480.png)
2. Imported the xml profile
3. Tried to add a channel  
  ![image](https://user-images.githubusercontent.com/1366654/136584342-c11f3393-a80f-4f02-a14d-b84f3e2dca9d.png)

I checked the commit where the support was added: https://github.com/oalt/fmc4se/commit/887a842ce94efb0e8b6e89ae2f46014013989a63. It shows the tags .222 and .223 only.

![image](https://user-images.githubusercontent.com/1366654/136584618-d4d36b8f-68ed-4ed9-818a-b1c5702d783f.png)

Maybe, I am not knowledgeable enough for Enterprise Architect and that I cannot see how version .221 supports the things committed for version .222 and .223.
